apps/functions-subscription/src/adapters/azure/functions/__tests__/get-activation-job.test.ts

```diff
@@ -0,0 +1,48 @@
+import { describe, expect, it } from 'vitest';
+import * as TE from 'fp-ts/lib/TaskEither';
+import * as O from 'fp-ts/lib/Option';
+import { makeFunctionContext, makeTestSystemEnv } from './mocks';
+import { makeAValidGetActivationJobRequest } from './data';
+import { anActivationJob } from '../../../../domain/__tests__/data';
+import { makeGetActivationJobHandler } from '../get-activation-job';
+
+describe('getActivationJob', () => {
+  it('should return 200 when the activation job exists', async () => {
+    const env = makeTestSystemEnv();
+
+    env.getActivationJob.mockReturnValueOnce(TE.right(O.some(anActivationJob)));
+
+    const actual = await makeGetActivationJobHandler(env)(
+      makeAValidGetActivationJobRequest(),
+      makeFunctionContext(),
+    );
+
+    const expectedJson = {
+      usersToActivate: anActivationJob.usersToActivate,
+      usersActivated: anActivationJob.usersActivated,
+      trialId: anActivationJob.trialId,
+    };
+
+    expect(actual.status).toStrictEqual(200);
+    expect(await actual.json()).toMatchObject(expectedJson);
```

Comment:
**question**: Given that the compiler will produce a type error if the structure does not match, do you think this check makes the code more robust?